File: Java/src/FizzBuzz.java
/*
Run with Java IDE
*/

import java.util.Scanner;

public class FizzBuzz {
    // Logic
    private static String divisible(int i){
        if (i%3 == 0 && i%5 == 0) {
            return "fizzBuzz";
        } else if (i%3 == 0){
            return "fizz";
        } else if (i%5 == 0){
            return "Buzz";
        } else {
            return String.valueOf(i);
        }
    }

    // Main
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        System.out.print("Please enter max number: ");
        int maxNum = scan.nextInt();
        // Loop
        for (int i = 1; i <= maxNum; i++){
            System.out.println(divisible(i));
        }
    }
}

File: Go/fizzbuzz.go
/*
Run with go run fizzbuzz.go
Compile with go build -o fizzbuzz
Run with ./fizzbuzz
*/

package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
)

// Logic
func divisible(i int) string{
	if i%3 == 0 && i%5 == 0{
        return "fizzBuzz"
	} else if i%3 == 0{
        return "fizz"
	} else if i%5 == 0{
        return "Buzz"
	} else {
    	return strconv.Itoa(i)
	}
}

// Main
func main() {
	scanner := bufio.NewScanner(os.Stdin)
	fmt.Print("Please enter max number: ")
	scanner.Scan()
	maxNum, _ := strconv.Atoi(scanner.Text())
	// Loop
	for i := 1; i <= maxNum; i++ {
		fmt.Println(divisible(i))
	}
}